src/test/java/plugins/OicAuthPluginTest.java

```diff
@@ -0,0 +1,307 @@
+package plugins;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.in;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThrows;
+
+import dasniko.testcontainers.keycloak.KeycloakContainer;
+import jakarta.inject.Inject;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.po.GlobalSecurityConfig;
+import org.jenkinsci.test.acceptance.po.LoggedInAuthorizationStrategy;
+import org.jenkinsci.test.acceptance.po.OicAuthSecurityRealm;
+import org.jenkinsci.test.acceptance.po.WhoAmI;
+import org.jenkinsci.test.acceptance.utils.keycloack.KeycloakUtils;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.resource.GroupResource;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.CredentialRepresentation;
+import org.keycloak.representations.idm.GroupRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.RoleRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.openqa.selenium.NoSuchElementException;
+
+@WithPlugins("oic-auth")
+public class OicAuthPluginTest extends AbstractJUnitTest {
+
+    private static final String REALM = "test-realm";
+    private static final String CLIENT = "jenkins";
+
+    @Rule
+    public KeycloakContainer keycloak = new KeycloakContainer();
```

Comment:
this uses the `latest` version of the container at time of running, which could lead to unexpected failures in PRs when a new version of keycloak is published.

Probably better to use a specific version here and get renovate to update it.

Reply:
https://github.com/jenkinsci/acceptance-test-harness/pull/1701/commits/d9fa1de575fa825f5b9c6fbef08873403c69472a